packages/mdc-textfield/index.js

```diff
@@ -109,12 +115,28 @@ class MDCTextField extends MDCComponent {
     if (outlineElement) {
       this.outline_ = outlineFactory(outlineElement);
     }
-    if (this.input_.hasAttribute(strings.ARIA_CONTROLS)) {
```

Comment:
The reason I removed relying on aria attributes is because of consistency with the way Character Counter is query selected. Since these 2 subcomponents are siblings I thought it makes more sense to use `nextElementSibling` to select either of the elements than using separate selectors.

Reply:
I completely understand we should aim to keep this subcomponent as abstract as possible so it can be moved to a shared component at some point of time.

Here are my other concerns of mandating `aria-controls` on input to make helper text work:

- `aria-controls` might not be appropriate for consumers use case. There are other relevant aria attribute such as `aria-labelledby` that consumer might use instead of `aria-controls`.
- `aria-controls` is required for text fields that has popup that shows autocomplete suggestions and it conflicts with current helper text structure.
- `aria-controls` is more appropriate when a particular element controlling a content such as popup, tabcontent, expand / collapse button etc.

Moreover, we're not changing any structure inside the subcomponent which might break the abstraction. We can may be change the layout of select component to keep it consistent with text field.

WDYT?

Reply:
RE the 2nd bullet, `aria-controls` can target multiple elements... but it sounds like the usability of it with screen readers is basically nonexistent in that case especially. And point taken RE one of the common use cases of helper text being static, in which case `aria-controls` doesn't even make sense really. I think we need to update more docs in the Accessibility section about this though.

Reply:
As discussed offline with Ken, documenting and using appropriate aria attributes requires some investigation which I'll handle separately.

Also this issue reported by Will seems related: #2583